src/libsyntax/parse/parser.rs

```diff
@@ -4671,7 +4677,6 @@ impl<'a> Parser<'a> {
                     bounds.push(TraitTyParamBound(poly_trait, modifier));
                 }
                 if has_parens {
-                    self.expect(&token::CloseDelim(token::Paren))?;
                     if let Some(&RegionTyParamBound(..)) = bounds.last() {
                         self.span_err(self.prev_span,
                                       "parenthesized lifetime bounds are not supported");
```

Comment:
Nit: Could you move this error into `if self.token.is_lifetime()` above as well?